src/coreComponents/codingUtilities/Table.hpp

```diff
@@ -0,0 +1,235 @@
+   * @tparam N
+   * @tparam Args
+   * @param args
+   */
+  template< size_t N, typename ... Args >
+  void addRow( Args const &... args );
+
+  /**
+   * @brief Write the the table into specified stream
+   * @param os
+   */
+  void draw( std::ostream & os = std::cout );
+
+  /**
+   * @brief Set the name of the table
+   * @param [in] title The name of the table
+   */
+  void setTitle( string_view title );
+
+  /**
+   * @brief Sets the minimal margin width between the row content an its borders.
+   * @param valueType
+   */
+  void setMargin( MarginValue valueType );
+
+  /**
+   * @return return the table name
+   */
+  string_view getTitle();
+
+private:
+
+  enum Section { header, values };
+
+  /**
+   * @brief Struct for a column.
+   */
+  struct Column
+  {
+    ColumnParam parameter;
+    // A vector containing all column values
+    std::vector< string > columnValues;
+    // The largest string in the column
+    string m_maxStringSize;
+  };
+
+  /**
+   * @brief Get the Margin object
+   * @param type
+   * @return Margin
+   */
+  Margin getMargin( MarginType type ) const;
+
+  /**
+   * @brief Get the name of the section given an enum
+   * @param section
+   * @return The name of the section
+   */0
+  string getStringSection( Section section ) const;
+
+  /**
+   * @brief fill the vector \p m_column with values from m_cellsRows who store all value in an unsorted order
+   */
+  void fillColumnsValuesFromMCellsRows();
+
+  /**
+   * @brief Split all header names by detecting \p '\n' character and store each split header name into a vector.
+   */
+  void splitHeadersStringAndStore();
+
+  /**
+   * @brief Iterate throught all header names vector and set them to the same size by adding empty string(s).
+   */
+  void addSpaceToSplitHeaderAndStore();
+
+  /**
+   * @brief For each column find the largest string size
+   */
+  void findMaxStringSize();
+
+  /**
+   * @brief If the title is the largest string size in the table, recalculate for all column, the \p m_maxStringSize value
+   * by adding an extra number of characters
+   */
+  void computeAndSetMaxStringSize( string::size_type sectionlineLength,
+                                   string::size_type titleLineLength );
+  /**
+   * @brief Compute the lines separator
+   */
+  void computeAndBuildLines();
+
+  /**
+   * @brief Build the title section
+   *
+   */
+  void buildTitleRow();
+
+  /**
+   * @brief build the header or values section
+   * @param nbRows
+   * @param sectionName
+   */
+  void buildSectionRows( integer nbRows, Section sectionName );
+
+  std::vector< std::vector< string > > m_splitHeader;
+  std::vector< std::vector< string > > m_cellsRows;
+
+  std::vector< Column > m_columns;
+
+  Margin borderMargin;
+  Margin columnMargin;
+
+  string title;
+  string topSeparator;
+  string sectionSeparator;
+  size_t maxRowHeader;
+
+  int marginTitle = 2;
+
+  string titleRow;
+  string rows;
```

Comment:
A lot of these values are just computed and used in the `draw()` method, is there a point to store them forever?
Some of these values may just be some input / output of your `buidX()` methods.